website/docs/d/eventhub_sas.html.markdown

```diff
@@ -0,0 +1,79 @@
+---
+subcategory: "Messaging"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_eventhub_sas"
+description: |-
+  Gets a Shared Access Signature (SAS Token) for an existing Event Hub.
+---
+
+# Data Source: azurerm_eventhub_sas
+
+Use this data source to obtain a Shared Access Signature (SAS Token) for an existing Event Hub.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_eventhub_namespace" "example" {
+  name                = "example-ehn"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  sku                 = "Basic"
+}
+
+resource "azurerm_eventhub" "example" {
+  name                = "example-eh"
+  namespace_name      = azurerm_eventhub_namespace.example.name
+  resource_group_name = azurerm_resource_group.example.name
+  partition_count     = 1
+  message_retention   = 1
+}
+
+resource "azurerm_eventhub_authorization_rule" "example" {
+  name                = "example-ehar"
+  namespace_name      = azurerm_eventhub_namespace.example.name
+  eventhub_name       = azurerm_eventhub.example.name
+  resource_group_name = azurerm_resource_group.example.name
+
+  listen = true
+  send   = true
+  manage = true
+}
+
+data "azurerm_eventhub_authorization_rule" "example" {
+  name                = azurerm_eventhub_authorization_rule.example.name
+  namespace_name      = azurerm_eventhub_namespace.example.name
+  eventhub_name       = azurerm_eventhub.example.name
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+data "azurerm_eventhub_sas" "example" {
+  connection_string = data.azurerm_eventhub_authorization_rule.example.primary_connection_string
+  expiry            = "2023-06-23T00:00:00Z"
+}
+
+output "sas_token" {
+  value     = replace(data.azurerm_eventhub_sas.test.sas, "\u0026", "&")
```

Comment:
how come we need this replace statement here?

Reply:
The generated sas token is encoded so that it includes "\u0026". But it cannot be directly leveraged by real scenario like sending message to eventhub. So I add this output here to provide better user experience. Below is an example of sending message to eventhub.

Note: The sas token that is generated by [azurerm_storage_account_sas](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/storage/storage_account_sas_data_source.go) also includes "\u0026".

Good Case:
```
POST https://acctest-ehn-test02.servicebus.windows.net/acctest-eh-test02/messages?timeout=60&api-version=2014-01

Content-Type: application/atom+xml;type=entry;charset=utf-8
Authorization: SharedAccessSignature sr=sb%3A%2F%2Facctest-ehn-test02.servicebus.windows.net%2Facctest-eh-test02&sig=ACAQ2RCBh85E3s%2A%2BCdiuOJv%3D&se=2568981021&skn=acctest-ehar-test02

{
    "DeviceId": "dev-cccc",
    "Temperature": "38.0"
}
```

Bad Case:
```
POST https://acctest-ehn-test02.servicebus.windows.net/acctest-eh-test02/messages?timeout=60&api-version=2014-01

Content-Type: application/atom+xml;type=entry;charset=utf-8
Authorization: SharedAccessSignature sr=sb%3A%2F%2Facctest-ehn-test02.servicebus.windows.net%2Facctest-eh-test02\u0026sig=A3s%2CeA%2BCdiuOJv%3D\u0026se=2568981021\u0026skn=acctest-ehar-test02

{
    "DeviceId": "dev-cccc",
    "Temperature": "38.0"
}
```

Reply:
@manicminer , after I did more testing, I found encoding the SAS token value is expected in the output block. Because the value would be encoded when "Sensitive: true," is added. However, the value wouldn't be encoded when it's directly referenced by another resource. So I will remove the output block since it's expected.